test/routes.js

```diff
@@ -655,3 +655,67 @@ test('GET graphql endpoint with prefixed wrapper', async (t) => {
 
   t.strictEqual(res.statusCode, 200)
 })
+
+test('Custom error handler', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      add(x: Int, y: Int): Int
+    }
+  `
+
+  const resolvers = {
+    add: async ({ x, y }) => {
+      throw new Error('dummy error')
+    }
+  }
+
+  function errorHandler (error, req, reply) {
+    app.log.error(error)
+    reply.code(403)
+    reply.send()
+  }
+
+  app.register(GQL, {
+    schema,
+    resolvers,
+    errorHandler
+  })
```

Comment:
Can you add a test that passes `errorHandler: false` and sets a root error handler in fastify?

Reply:
I think we do not see this option the same way, if I set `errorHandler` to false, based on the code https://github.com/mcollina/fastify-gql/pull/28/files#diff-d640e5835a6b0c342cf705a06018c2b9R70 it will enable the default error handler of `fastify-gql`.

I can change this if you prefer to have an option that enables/disable the fastify-gql errorHandler and having the user define it in his scope. The way I've implemented it here I was passing the handler function to fastify-gql.

Reply:
Commit is ready to match the behavior you asked (errorHandler is only a Boolean that disable or enable the fastify-gql error handler). Let me know if it's really what you requested and I'll push it.